yt/frontends/amrvac/fields.py

```diff
@@ -59,64 +64,89 @@ def _get_ekin(data):
+            v3 = data["amrvac", "m3"] / data["amrvac", "rho"]
+            v3.units = self.ds.unit_system['velocity']
+            return v3
+
         def _total_energy(field, data):
             etot = _get_ekin(data)
             if ("amrvac", "b1") in self.field_list:
                 etot = etot + _get_emag(data)
+            etot.units = self.ds.unit_system['specific_energy']
             return etot
 
         # pressure field, add this first to calculate temperature after
         self.add_field(("amrvac", "pressure"), sampling_type="cell",
-                       function=_pressure, units="")
-        self.alias(("gas", "pressure"), ("amrvac", "pressure"), units="")
+                       function=_pressure, units=code_pressure)
+        self.alias(("gas", "pressure"), ("amrvac", "pressure"), units=self.ds.unit_system['pressure'])
+
         # temperature field
         self.add_field(("amrvac", "temperature"), sampling_type="cell",
-                       function=_temperature, units="")
-        self.alias(("gas", "temperature"), ("amrvac", "temperature"), units="")
+                       function=_temperature, units=code_temperature)
+        self.alias(("gas", "temperature"), ("amrvac", "temperature"), units=self.ds.unit_system['temperature'])
+
         # velocity fields
         self.add_field(("amrvac", "velocity_1"), sampling_type="cell",
-                       function=_velocity1, units="")
-        self.alias(("amrvac", "v1"), ("amrvac", "velocity_1"), units="")
+                       function=_velocity1, units=code_velocity)
+        self.alias(("amrvac", "v1"), ("amrvac", "velocity_1"), units=self.ds.unit_system['velocity'])
         if ("amrvac", "m2") in self.field_list:
             self.add_field(("amrvac", "velocity_2"), sampling_type="cell",
-                           function=_velocity2, units="")
-            self.alias(("amrvac", "v2"), ("amrvac", "velocity_2"), units="")
+                           function=_velocity2, units=code_velocity)
+            self.alias(("amrvac", "v2"), ("amrvac", "velocity_2"), units=self.ds.unit_system['velocity'])
         if ("amrvac", "m3") in self.field_list:
             self.add_field(("amrvac", "velocity_3"), sampling_type="cell",
-                           function=_velocity3, units="")
-            self.alias(("amrvac", "v3"), ("amrvac", "velocity_3"), units="")
+                           function=_velocity3, units=code_velocity)
+            self.alias(("amrvac", "v3"), ("amrvac", "velocity_3"), units=self.ds.unit_system['velocity'])
+
         # total energy
         self.add_field(("amrvac", "total_energy"), sampling_type="cell",
-                       function=_total_energy, units="")
-        self.alias(("gas", "total_energy"), ("amrvac", "total_energy"), units="")
+                       function=_total_energy, units=code_energy)
+        self.alias(("gas", "total_energy"), ("amrvac", "total_energy"), units=self.ds.unit_system['specific_energy'])
```

Comment:
not sure about this. I do not trust that the "energy" field should have the dimension of a specific energy (energy/mass). Can you check and confirm ?